packages/sol/src/error.ts

```diff
@@ -1,7 +1,25 @@
+import { ApiErrorCode } from "@litespace/types";
 export async function safe<T>(handler: () => Promise<T>) {
   try {
     return await handler();
   } catch (error) {
     return error instanceof Error ? error : new Error("unkown");
   }
 }
+
+type CodedError = {
+  message: string;
+  code: ApiErrorCode;
+};
+
+// api error handling
+export class ResponseError extends Error {
+  statusCode: number;
+  errorCode: ApiErrorCode;
+
+  constructor(error: CodedError, statusCode: number) {
```

Comment:
A better interface will look something like this: 

```suggestion
  constructor({errorCode, statusCode, message}: {
    message: string;
    errorCode: ApiErrorCode;
    statusCode: number;
  }) {
```